mopidy_tidal/backend.py

```diff
@@ -71,13 +92,12 @@ def on_start(self):
             _connecting_log("using default client id & client secret from python-tidal")
 
         self._active_session = Session(config)
-        if not self._config["tidal"]["lazy"]:
+        if not self.lazy_connect:
             self._login()
 
     def _login(self):
         # Always store tidal-oauth cache in mopidy core config data_dir
-        data_dir = Extension.get_data_dir(self._config)
-        oauth_file = os.path.join(data_dir, "tidal-oauth.json")
+        oauth_file = self.data_dir / "tidal-oauth.json"
```

Comment:
Is this syntax preferred over os.path.join?

Reply:
well it is by me... the difference is that `self.data_dir` is now a `Path()` object, so we can use pathlib stuff on it. One of those is the overriden `__div__` so you can write paths out with `/`.  I cherry-picked [this refactor](https://github.com/tehkillerbee/mopidy-tidal/pull/116/commits/e30aabd74920bb9a2bade9710ff39ab876d7d8a7) from a stalled PR improving the cache, I think because I needed to make/delete dirs and it was easier with pathlib.

But I personally think the pathlib syntax of `path_obj / path_obj / path_obj / "str2"` is neat, and easier to read than `os.path.join`.  (There's no decent os.path replacement I know of for `Path().mkdir(exist_ok=True, parents=True)`, so we were actually casting strs to `Path`s every time we needed to make a directory anyhow.)  The other pathlib way of doing it is `Path(path_obj, path_obj, path_obj, "str", "str")`, but that's too magical in my book---a newcomer isn't going to know that the constructor basically calls `os.path.join` on `*args`, so I try to avoid using it, even preferring string interpolation.  Whereas `/` is so natural you don't even notice it's doing something clever until you think about it.

Thoughts?

Reply:
> But I personally think the pathlib syntax of path_obj / path_obj / path_obj / "str2" is neat, and easier to read than os.path.join

Agreed, its easier to read the path when written in that way. Its probably also a better idea to use a Path() object instead.